src/web/components/ApiKeyManagement/KeyCreationDialog.tsx

```diff
@@ -0,0 +1,130 @@
+import { useState } from 'react';
+import { SubmitHandler } from 'react-hook-form';
+
+import { ApiRoleDTO } from '../../../api/entities/ApiRole';
+import { ApiKeySecretsDTO } from '../../../api/services/apiKeyService';
+import { CreateApiKeyFormDTO } from '../../services/apiKeyService';
+import { Dialog } from '../Core/Dialog';
+import { Form } from '../Core/Form';
+import KeySecretReveal from '../Core/KeySecretReveal';
+import { StatusPopup } from '../Core/StatusPopup';
+import { CheckboxInput } from '../Input/CheckboxInput';
+import { TextInput } from '../Input/TextInput';
+
+import './KeyCreationDialog.scss';
+
+type KeySecretProp = ApiKeySecretsDTO | undefined;
+
+type KeyCreationDialogProps = {
+  onKeyCreation: (form: CreateApiKeyFormDTO) => Promise<KeySecretProp>;
+  triggerButton: JSX.Element;
+  availableRoles: ApiRoleDTO[];
+};
+
+function KeyCreationDialog({
+  onKeyCreation,
+  triggerButton,
+  availableRoles,
+}: KeyCreationDialogProps) {
+  const [open, setOpen] = useState(false);
+  const [secrets, setSecrets] = useState<KeySecretProp>(undefined);
+  const [showStatusPopup, setShowStatusPopup] = useState<boolean>(false);
+  const [statusPopupMessage, setStatusPopupMessage] = useState<string>('');
+  const [copiedSecrets, setCopiedSecrets] = useState<Map<String, boolean>>(
+    new Map<String, boolean>()
+  );
+
+  const onFormSubmit: SubmitHandler<CreateApiKeyFormDTO> = async (formData) => {
+    setSecrets(await onKeyCreation(formData));
+    setStatusPopupMessage(
+      'Your key has been created. Please copy the credentials before closing the window.'
+    );
+    setShowStatusPopup(true);
+  };
+
+  function confirmClose(): void {
+    const values = [...copiedSecrets.values()];
+
+    if (values.filter((value) => value === false).length > 0) {
+      setStatusPopupMessage('Please copy all secrets shown before closing the page');
```

Comment:
This message should be associated warning or info, rather than success.

Reply:
Makes sense.